### PR TITLE
sys/net/sock: add sock_aux_rssi

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

- Add module `sock_aux_rssi` to be able to access RSSI values via the SOCK API
- Implement this pseudio module for `gnrc_sock_udp` and `gnrc_sock_ip` (which will also work for DTLS than)
- Extend `tests/sock_gnrc_udp` and `tests/sock_gnrc_ip` to also tests RSSI values
- Add `tests/sock_udp_aux`, which is a UDP echo server that prints all currently supported auxiliary data about received UDP datagrams and sent replies

### Testing procedure

- Murdock will run the unit tests in `tests/sock_gnrc_udp` and `tests/sock_gnrc_ip` for you
- The test application in `tests/sock_udp_aux` will allow testing the RSSI value on real hardware

### Issues/PRs references

- [x] Depends on and contains: https://github.com/RIOT-OS/RIOT/pull/15754

### CODE REVIEW

File: tests/sock_udp_aux/main.c [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+#include "fmt.h"
+#include "net/sock/udp.h"
+#include "shell.h"
+#include "shell_commands.h"
+#include "net/ipv6/addr.h"
+
+#define MAIN_QUEUE_SIZE     (8)
+
+#define PORT                12345
```
Comment (benpicco) [inline tests/sock_udp_aux/main.c:32]:
```suggestion
#define PORT_UDP            12345
```

to avoid conflicts with vendor files

**Comment by maribu** [comment]
Fixed the conflict as suggested, also fixed a comment, and squashed.

**Review by benpicco — APPROVED** [review]

**Comment by maribu** [comment]
For reference, this is the failed test in the CI:

```
-- running on worker mobi6.inet.haw-hamburg.de thread 1, build number 86587.
make: Entering directory '/tmp/dwq.0.9261969958819269/b17fff2444e901695ff00183c078dd03/tests/thread_flags'
Building application "tests_thread_flags" for "native" with MCU "native".

sha1sum /tmp/dwq.0.9261969958819269/b17fff2444e901695ff00183c078dd03/tests/thread_flags/tests/01-run.py /tmp/dwq.0.9261969958819269/b17fff2444e901695ff00183c078dd03/build/tests_thread_flags.elf > /tmp/dwq.0.9261969958819269/b17fff2444e901695ff00183c078dd03/build/test-input-hash.sha1
   text	   data	    bss	    dec	    hex	filename
  36991	    596	  60124	  97711	  17daf	/tmp/dwq.0.9261969958819269/b17fff2444e901695ff00183c078dd03/build/tests_thread_flags.elf
make: Leaving directory '/tmp/dwq.0.9261969958819269/b17fff2444e901695ff00183c078dd03/tests/thread_flags'
make: Entering directory '/tmp/dwq.0.9261969958819269/b17fff2444e901695ff00183c078dd03/tests/thread_flags'
r
/tmp/dwq.0.9261969958819269/b17fff2444e901695ff00183c078dd03/build/tests_thread_flags.elf /dev/ttyACM0 
RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

Help: Press s to start test, r to print it is ready
READY
s
START
main(): This is RIOT! (Version: buildtest)
START
thread(): waiting for 0x1...
main(): setting flag 0x0001
thread(): received flags: 0x0001
thread(): waiting for 0x1 || 0x64...
main(): setting flag 0x0064
thread(): received flags: 0x0064
thread(): waiting for 0x2 && 0x4...
main(): setting flag 0x0001
main(): setting flag 0x0008
main(): setting flag 0x0002
main(): setting flag 0x0004
thread(): received flags: 0x0006
thread(): waiting for any flag, one by one
thread(): received flags: 0x0001
thread(): waiting for any flag, one by one
thread(): received flags: 0x0008
main: setting 100ms timeout...
main: timeout triggered. time passed: 102870us
main: setting 100ms timeout (using uint64)...
main: timeout triggered. time passed: 103824us
FAILURE
Timeout in expect script at "child.expect("SUCCESS")" (tests/thread_flags/tests/01-run.py:29)

/tmp/dwq.0.9261969958819269/b17fff2444e901695ff00183c078dd03/tests/thread_flags/../../Makefile.include:781: recipe for target 'test' failed
make: *** [test] Error 1
make: Leaving directory '/tmp/dwq.0.9261969958819269/b17fff2444e901695ff00183c078dd03/tests/thread_flags'
-- build directory size: 884K
```

I'd say this is unrelated, which is even more worrisome. I don't have time to investigate now, but I hope I'll be able to take a look tomorrow.

File: tests/sock_udp_aux/Makefile.ci [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    nucleo-f042k6 \
+    nucleo-f303k8 \
+    nucleo-f334r8 \
+    nucleo-f410rb \
+    nucleo-l011k4 \
+    nucleo-l031k6 \
+    nucleo-l053r8 \
+    samd10-xmini \
+    slstk3400a \
+    stk3200200 \
```
Comment (benpicco) [inline tests/sock_udp_aux/Makefile.ci:28]:
```suggestion
    stk3200 \
```

Comment (maribu) [reply tests/sock_udp_aux/Makefile.ci]: A typical vim user error :-D

**Comment by maribu** [comment]
All green :-)

**Comment by maribu** [comment]
Thanks :-)

**Pull request merged**